### ISSUE TITLE
Question about Neighbor List (nei_list) Generation and Pedestrian Isolation

### ISSUE THREAD

**Issue by WangYLon** [Majiker/STAR#30]
Dear Authors,  

Thank you for sharing your excellent work, which has been a great source of inspiration for me. However, I have a question regarding the generation of `nei_list` during data processing, and I hope you can help clarify my doubts.  

Below is the relevant portion of your code for generating `nei_list`:  

```python
def get_social_inputs_numpy(self, inputnodes):
    """
    Get the sequence list (denoting where data exists) and neighboring list (denoting where neighbors exist).
    """
    # inputsnodes: (seq_len, num_Peds, 2)
    num_Peds = inputnodes.shape[1]

    seq_list = np.zeros((inputnodes.shape[0], num_Peds))  # (T, num_Peds)
    # denote where data not missing
    for pedi in range(num_Peds):
        seq = inputnodes[:, pedi]  # (T,2)
        seq_list[seq[:, 0] != 0, pedi] = 1  # 1 means data exists

    # get relative cords, neighbor id list
    nei_list = np.zeros((inputnodes.shape[0], num_Peds, num_Peds))  # (T, N, N)
    nei_num = np.zeros((inputnodes.shape[0], num_Peds))  # (T, N)

    # nei_list[f, i, j] denotes if j is i's neighbor in frame f
    for pedi in range(num_Peds):
        nei_list[:, pedi, :] = seq_list
        nei_list[:, pedi, pedi] = 0  # person i is not the neighbor of itself
        nei_num[:, pedi] = np.sum(nei_list[:, pedi, :], 1)
        seqi = inputnodes[:, pedi]
        for pedj in range(num_Peds):
            seqj = inputnodes[:, pedj]
            select = (seq_list[:, pedi] > 0) & (seq_list[:, pedj] > 0)

            relative_cord = seqi[select, :2] - seqj[select, :2]

            # invalid data index
            select_dist = (abs(relative_cord[:, 0]) > self.args.neighbor_thred) | (
                abs(relative_cord[:, 1]) > self.args.neighbor_thred
            )

            nei_num[select, pedi] -= select_dist

            select[select == True] = select_dist
            nei_list[select, pedi, pedj] = 0
    return seq_list, nei_list, nei_num
```

In pedestrian trajectory prediction tasks, a sequence of 20 frames is commonly treated as a scene (representing either a complete trajectory or a time window). My understanding is that pedestrians in different scenes should be strictly isolated to avoid data leakage. This concept is also reflected in your `mean_normalize_abs_input` function, as shown below:  

```python
def mean_normalize_abs_input(self, node_abs, st_ed):
    """
    :param node_abs: Absolute coordinates of pedestrians
    :type Tensor
    :param st_ed: List of tuples indicating the indices of pedestrians belonging to the same scene
    :type List of tuples
    :return: node_abs: Normalized absolute coordinates of pedestrians
    :rtype: Tensor
    """
    node_abs = node_abs.permute(1, 0, 2)
    node_abs_c = node_abs.clone()
    for st, ed in st_ed:
        mean_x = torch.mean(node_abs[st:ed, :, 0])
        mean_y = torch.mean(node_abs[st:ed, :, 1])

        node_abs_c[st:ed, :, 0] = node_abs[st:ed, :, 0] - mean_x
        node_abs_c[st:ed, :, 1] = node_abs[st:ed, :, 1] - mean_y

    return node_abs_c.permute(1, 0, 2)
```

However, when reviewing the generation of `nei_list`, it appears that pedestrians in the same batch (potentially from different scenes) are considered neighbors. This seems to conflict with the principle of pedestrian isolation between different scenes.  

Could you please confirm if this understanding is correct? If so, could you elaborate on why this design was chosen, and whether it might inadvertently affect the accuracy or fairness of the model?  

Thank you for your time and assistance. I look forward to your insights!  

Best regards,
